sdks/java/core/src/main/java/org/apache/beam/sdk/metrics/MetricResult.java

```diff
@@ -32,14 +33,13 @@
 
   /**
    * Return the value of this metric across all successfully completed parts of the pipeline.
-   *
-   * <p>Not all runners will support committed metrics. If they are not supported, the runner will
-   * throw an {@link UnsupportedOperationException}.
    */
+  @Nullable
```

Comment:
I don't think we should change this, maybe better to enforce that the runners throw the exception also because of making type sense for that T.

Reply:
Both the direct runner and dataflow runner use autovalue to implement this class and return null rather then throwing an exception. It seems changing those implementations to match the documentation has a bigger potential to break users then changing the documentation to match the implementation. Are there other implementations I missed?

Reply:
Looks like I wasn't looking hard enough. There are implementations that obey this contract. I'll update dataflow instead of changing it.